common/src/uds_fe_programming.rs

```diff
@@ -0,0 +1,187 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    uds_fe_programming.rs
+
+Abstract:
+
+    File contains the implementation of UDS/FE programming flow that can be
+    used by both ROM and Runtime.
+
+--*/
+
+use crate::cprintln;
+use caliptra_drivers::{AxiAddr, CaliptraError, CaliptraResult, Dma, DmaOtpCtrl, SocIfc, Trng};
+
+/// UDS/FE Programming Flow
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum UdsFeProgrammingFlow {
+    /// UDS (Unique Device Secret) programming mode - 64 bytes
+    Uds,
+    /// FE (Field Entropy) programming mode - 32 bytes, can be set with a u32 granularity using bitmask
+    Fe { bitmask: u32 },
+}
+
+impl UdsFeProgrammingFlow {
+    /// Returns true if this is UDS programming mode
+    pub fn is_uds(self) -> bool {
+        matches!(self, UdsFeProgrammingFlow::Uds)
+    }
+
+    /// Returns the seed length in 32-bit words for this mode
+    pub fn seed_length(self) -> usize {
+        match self {
+            UdsFeProgrammingFlow::Uds => 16, // 64 bytes = 16 u32 words
+            UdsFeProgrammingFlow::Fe { bitmask: _ } => 8, // 32 bytes = 8 u32 words
+        }
+    }
+
+    /// Returns the prefix string for logging
+    pub fn prefix(self) -> &'static str {
+        match self {
+            UdsFeProgrammingFlow::Uds => "uds",
+            UdsFeProgrammingFlow::Fe { bitmask: _ } => "fe",
+        }
+    }
+
+    /// Returns the destination address for programming
+    fn get_dest_address(&self, soc_ifc: &SocIfc) -> u32 {
+        match self {
+            Self::Uds => soc_ifc.uds_seed_dest_base_addr_low(),
+            // [CAP2][TODO] This needs to be different for field entropy
+            Self::Fe { bitmask: _ } => soc_ifc.uds_seed_dest_base_addr_low(),
+        }
+    }
+
+    /// Programs either UDS (64 bytes) or FE (32 bytes) based on the enum variant
+    ///
+    /// # Arguments
+    /// * `soc_ifc` - SoC interface for accessing fuse controller and related registers
+    /// * `trng` - TRNG for generating random seeds
+    /// * `dma` - DMA engine for OTP control
+    pub fn program_uds_fe(
```

Comment:
nit: I think we could just name this `program` to make it scalable in case we decide to add more fuse programming to this enum.